### PR TITLE
Add a whiteout tool as the secondary function of the pencil tool

### PR DESCRIPTION
**Pull request by finnboeger**
This allows easy touch ups without using the masking eraser. This works best on a white canvas - when using the grid the whiteout spots can be seen.
After #45 has been merged this can be modified to use a different size for the whiteout tool (e.g. twice the current pen size).
After #46 has been merged this should display a marker of the proper size.

### CODE REVIEW

**Comment by lovasoa** [comment]
Yes, the eraser should definitely not erase the grid... Also, this should probably not be merged with the pencil...

**Comment by finnboeger** [comment]
This is not an eraser but rather a whiteout tool (like [Tipp-Ex](https://en.wikipedia.org/wiki/Tipp-Ex)) which does overwrite everything below it.

File: server/sockets.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ var iolib = require('socket.io')
 	, log = require("./log.js").log
 	, BoardData = require("./boardData.js").BoardData;
 
-var MAX_EMIT_COUNT = 64; // Maximum number of draw operations before getting banned
-var MAX_EMIT_COUNT_PERIOD = 5000; // Duration (in ms) after which the emit count is reset
+var MAX_EMIT_COUNT = 200; // Maximum number of draw operations before getting banned
+var MAX_EMIT_COUNT_PERIOD = 1000; // Duration (in ms) after which the emit count is reset
```
Comment (lovasoa) [inline server/sockets.js:6]: I think this is not related to this PR..

**Comment by lovasoa** [comment]
OK for the merging, then...

File: server/sockets.js (hunk 1 above)
Comment (finnboeger) [reply server/sockets.js]: This PR increases the polling rate of the pencil tool leading to a much smoother experience but also an increased message count. I just did some testing and found that I can't exceed 200 messages over 5 seconds unless I actively try. Even attempting to fill out a larger area by rapidly whacking the pencil left and right tends to not exceed that number.

File: client-data/tools/pencil/pencil.js [hunk 2]
```diff
@@ -48,19 +76,19 @@
 		Tools.drawAndSend({
 			'type': 'line',
 			'id': curLineId,
-			'color': Tools.getColor(),
+			'color': (curPen === "pencil" ? Tools.getColor() : "white"),
```
Comment (lovasoa) [inline client-data/tools/pencil/pencil.js:79]:
```suggestion
			'color': (curPen === "pencil" ? Tools.getColor() : "#ffffff"),
```

for consistency

File: client-data/tools/pencil/pencil.js [hunk 3]
```diff
@@ -38,6 +43,29 @@
...
+				cursor.remove();
+			}
+		}
+	}
+
+	function onStart(){
+		if(curPen === "whiteout"){
+			Tools.showMarker = true;
+		}
+	}
```
Comment (lovasoa) [inline client-data/tools/pencil/pencil.js:61]: I don't think the cursor should be accessed from other tools. You can use the `showMarker` attribute of individual tools that you introduced in the cursor PR. You can use something like `pencilTool.showMarker = False`

Comment (finnboeger) [reply client-data/tools/pencil/pencil.js]: I have decided that since we were displaying the cursor anyway (`alwaysShowMarker` was set to `true`) we don't need any of the code that was checking if it should hide the cursor

**Comment by lovasoa** [comment]
@finnboeger : I made a few changes. If you're okay with them, I'll merge.

**Comment by lovasoa** [comment]
OK, I'll merge, let me know if you have any comment.

**Comment by finnboeger** [comment]
Looks great :+1: 

Sorry for the late response, unfortunately I didn't have time in the last week.
